Fix for migration error

Parables opened the pull request:
SQLSTATE[42000]: Syntax error or access violation: 1072 Key column 'user_id' doesn't exist in table (SQL: alter table `social_providers` add constraint `social_providers_user_id_foreign` foreign key (`user_id`) references `users` (`id`) on delete cascade)

joselfonseca commented:
Thanks for the PR, I am not sure I understand the reason to delete the user id from the table as well as the foreign key. Also the checks are not passing. Can you please elaborate on the issue that led you to propose the PR?

Thanks.

joselfonseca commented on migrations/2019_11_19_000000_update_social_provider_users_table.php:
```diff
@@ -20,11 +20,15 @@ public function up()
-            $table->foreign('user_id')->references('id')->on('users')->onDelete('cascade');
+        });
+
+        // Fix for migration error:
```
no need for this comment

joselfonseca commented on migrations/2019_11_19_000000_update_social_provider_users_table.php:
```diff
@@ -20,11 +20,15 @@ public function up()
+        // Fix for migration error:
+        /*  SQLSTATE[42000]: Syntax error or access violation: 1072 Key column 'user_id' doesn't exist in table (SQL: alter table `social_providers` add constraint `social_providers_user_id_foreign` foreign key (`user_id`) references `users` (`id`) on delete cascade) */
+        Schema::table('social_providers', function (Blueprint $table) {
+            $table->foreignId('user_id')->after('provider_id')->constrained()->onDelete('cascade');;
```
This can be moved into the main create method.

Parables commented:
Thanks for replying back. On a fresh installation of Laravel and your
package, running `php artisan migrate ` throws the following error
```
SQLSTATE[42000]: Syntax error or access violation: 1072 Key column
'user_id' doesn't exist in table (SQL: alter table social_providers add
constraint social_providers_user_id_foreign foreign key (user_id)
references users (id) on delete cascade)
```

So further digging into the problem before filing an issue lead me to
discover that the foreign key constraint there is causing the migration to
fail so I simply rewrote that part like this:
```

// Fix for migration error:
+        /*  SQLSTATE[42000]: Syntax error or access violation: 1072
Key column 'user_id' doesn't exist in table (SQL: alter table
`social_providers` add constraint `social_providers_user_id_foreign`
foreign key (`user_id`) references `users` (`id`) on delete cascade)
*/
+        Schema::table('social_providers', function (Blueprint $table) {
+            $table->foreignId('user_id')->after('provider_id')->constrained()->onDelete('cascade');;

```

I have resolved all the requested changes and tested it out and it works.
Awaiting your approval.

Thanks again for this amazing package. Made my life so easy.


P.S: I would appreciate some help on the email verification part. When I
pass in my access token got from registering a user:
```mutation{
  register(input:{
    name: "Parables"
    email: ***@***.***"
    password: "password"
    password_confirmation: "password"
  }){
    tokens{
      access_token
      refresh_token
      expires_in
      token_type
      user{
        id
        name
        email
      }
    }
    status
  }
}
```

into this mutation:
```

  mutation {
      verifyEmail(input: {
          token:
"def502009bf0fcd0a80e511b027646356adb728ea60ad041df18cec39d6be24e19d224a3b080bb8e746361d1973b8264489a00a820434cb45e3d79587d1572b96a46306cf5ea0f730dc0b8ec5c070689eeb690ce5175781f07f6812d15f9e67ac3459e1aa13756380cd602177e7f65a9a4258e9e863f6367c86b3e8b140663cadf100f51b71e92dea10a3357f6d352e62e0bba0753ff23288b4ce9e276555154ea1d880f1d7826b9791df49b757ad58afff8721cab3062f8801a385f666678387013b1ee380f54a08dd63bbab122ab15c52a49d17752f3386deef3146cab8dc942e5c33649da8bc3b8f2fa17e39d4a15fdb078ad4ba1d02833280ea839b5c379920762ce5a38ab529eef9294f77f74bef8ee8e154142bbfc4f0bdd9720aa0123305c3401d9343caf6efa3d79ff4370ef8a358d2f4c2ad8890dcae721c50bc71defff6c9c67d4461071da76b9e70c573c1ef426eadd6ada9b3beaf8d871ade93967"
      }) {
          access_token
      }
  }
```

I get this error:
```
{
      "debugMessage": "Attempt to read property \"expiration\" on null",
      "message": "Internal server error",
...
```

Tried with booth login and registration access token and refresh token,
None works. Came to see if an actual verification email was sent which
might contain that token but didn't find anything thing here but your mail

On Sun, Jun 13, 2021 at 1:21 AM Jose Luis Fonseca ***@***.***>
wrote:

> Thanks for the PR, I am not sure I understand the reason to delete the
> user id from the table as well as the foreign key. Also the checks are not
> passing. Can you please elaborate on the issue that led you to propose the
> PR?
>
> Thanks.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/joselfonseca/lighthouse-graphql-passport-auth/pull/148#issuecomment-860134606>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJDDF3TVPR3TWILNR5S55SLTSQBY5ANCNFSM46TE3DJA>
> .
>

joselfonseca commented:
Thanks for addressing the feedback @Parables Unfortunately the check for PHP 8 Laravel 6 is not passing 

https://github.com/joselfonseca/lighthouse-graphql-passport-auth/pull/148/checks?check_run_id=2820018279

The method foreignId was introduced in a later version of laravel and is not compatible with Laravel 6. The reason why we are currently supporting laravel 6 is because is the current LTS.

https://laravel.com/docs/8.x/releases#support-policy

So we need to keep it compatible. Thanks for the PR and I will take a look at the issue you mention to see if it is present in a fresh installation. If you want to update your PR to support Laravel 6 please go ahead and I will take a look again. Thanks!

Parables commented:
Thank you so much for this amazing package and its a great honour for me to
contribute to it. I will do as you have said and report back to you


On Mon, Jun 14, 2021 at 1:11 PM Jose Luis Fonseca ***@***.***>
wrote:

> Thanks for addressing the feedback @Parables <https://github.com/Parables>
> Unfortunately the check for PHP 8 Laravel 6 is not passing
>
>
> https://github.com/joselfonseca/lighthouse-graphql-passport-auth/pull/148/checks?check_run_id=2820018279
>
> The method foreignId was introduced in a later version of laravel and is
> not compatible with Laravel 6. The reason why we are currently supporting
> laravel 6 is because is the current LTS.
>
> https://laravel.com/docs/8.x/releases#support-policy
>
> So we need to keep it compatible. Thanks for the PR and I will take a look
> at the issue you mention to see if it is present in a fresh installation.
> If you want to update your PR to support Laravel 6 please go ahead and I
> will take a look again. Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/joselfonseca/lighthouse-graphql-passport-auth/pull/148#issuecomment-860672742>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJDDF3TGIM5SPKVUEN22KYDTSX5YFANCNFSM46TE3DJA>
> .
>

Parables commented:
Apparently, it looks that the order of migrations on my end was not in
order causing the social_provider_users_table to execute before the users
migration was actually done

I don't think this is an actual bug in your package but it can lead to
frustration for new developers.

I suggest that we can check if the users table exists before setting up a
foreign key constraint or simply create it if it doesn't exist and set the
foreign key constraint

On Mon, Jun 14, 2021 at 5:14 PM Parables Boltnoel ***@***.***>
wrote:

> Thank you so much for this amazing package and its a great honour for me
> to contribute to it. I will do as you have said and report back to you
>
>
> On Mon, Jun 14, 2021 at 1:11 PM Jose Luis Fonseca <
> ***@***.***> wrote:
>
>> Thanks for addressing the feedback @Parables
>> <https://github.com/Parables> Unfortunately the check for PHP 8 Laravel
>> 6 is not passing
>>
>>
>> https://github.com/joselfonseca/lighthouse-graphql-passport-auth/pull/148/checks?check_run_id=2820018279
>>
>> The method foreignId was introduced in a later version of laravel and is
>> not compatible with Laravel 6. The reason why we are currently supporting
>> laravel 6 is because is the current LTS.
>>
>> https://laravel.com/docs/8.x/releases#support-policy
>>
>> So we need to keep it compatible. Thanks for the PR and I will take a
>> look at the issue you mention to see if it is present in a fresh
>> installation. If you want to update your PR to support Laravel 6 please go
>> ahead and I will take a look again. Thanks!
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/joselfonseca/lighthouse-graphql-passport-auth/pull/148#issuecomment-860672742>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AJDDF3TGIM5SPKVUEN22KYDTSX5YFANCNFSM46TE3DJA>
>> .
>>
>

Parables commented:
PR created. This pull might be bloated but well it does get rid of any
error.

I create the users table if it doesn't exist and add the avatar column but
if it exists, I add the avatar column to the existing users table.

The social_providers table is reordered with the user_id foreign key placed
above the timestamps

On Mon, Jun 14, 2021 at 6:03 PM Parables Boltnoel ***@***.***>
wrote:

> Apparently, it looks that the order of migrations on my end was not in
> order causing the social_provider_users_table to execute before the users
> migration was actually done
>
> I don't think this is an actual bug in your package but it can lead to
> frustration for new developers.
>
> I suggest that we can check if the users table exists before setting up a
> foreign key constraint or simply create it if it doesn't exist and set the
> foreign key constraint
>
> On Mon, Jun 14, 2021 at 5:14 PM Parables Boltnoel ***@***.***>
> wrote:
>
>> Thank you so much for this amazing package and its a great honour for me
>> to contribute to it. I will do as you have said and report back to you
>>
>>
>> On Mon, Jun 14, 2021 at 1:11 PM Jose Luis Fonseca <
>> ***@***.***> wrote:
>>
>>> Thanks for addressing the feedback @Parables
>>> <https://github.com/Parables> Unfortunately the check for PHP 8 Laravel
>>> 6 is not passing
>>>
>>>
>>> https://github.com/joselfonseca/lighthouse-graphql-passport-auth/pull/148/checks?check_run_id=2820018279
>>>
>>> The method foreignId was introduced in a later version of laravel and is
>>> not compatible with Laravel 6. The reason why we are currently supporting
>>> laravel 6 is because is the current LTS.
>>>
>>> https://laravel.com/docs/8.x/releases#support-policy
>>>
>>> So we need to keep it compatible. Thanks for the PR and I will take a
>>> look at the issue you mention to see if it is present in a fresh
>>> installation. If you want to update your PR to support Laravel 6 please go
>>> ahead and I will take a look again. Thanks!
>>>
>>> —
>>> You are receiving this because you were mentioned.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/joselfonseca/lighthouse-graphql-passport-auth/pull/148#issuecomment-860672742>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AJDDF3TGIM5SPKVUEN22KYDTSX5YFANCNFSM46TE3DJA>
>>> .
>>>
>>

joselfonseca commented on migrations/2019_11_19_000000_update_social_provider_users_table.php:
```diff
@@ -13,18 +13,32 @@ class UpdateSocialProviderUsersTable extends Migration
-        Schema::table('users', function (Blueprint $table) {
-            if (! Schema::hasColumn('users', 'avatar')) {
+        if (! Schema::hasTable('users')) {
+            Schema::create('users', function (Blueprint $table) {
```
We should not create the users table as that is not something this package's scope. If your migration has a post date you can just fix the timestamp in the users migration. Usually that is the first migration that should run, that is why this package asumes that the table already exists.

joselfonseca commented:
@Parables Thanks for all the attention to this PR, I am still not sure where the issue is. I'll try to do a fresh installation and see if I run into any issues.

Parables commented:
I really appreciate your time and support. This is not an actual issue with
this package so please feel free to close this issue.

One more thing, please can you provide a guide on how to get email
verification working just like you have a tutorial for auth.

I first used Laravel in V.5 but I didn't continue to learn it until now so
I'm not that experienced with the Laravel Ecosystem.

I believe it will be helpful to beginners like me.

On Tue, Jun 15, 2021, 9:44 PM Jose Luis Fonseca ***@***.***>
wrote:

> @Parables <https://github.com/Parables> Thanks for all the attention to
> this PR, I am still not sure where the issue is. I'll try to do a fresh
> installation and see if I run into any issues.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/joselfonseca/lighthouse-graphql-passport-auth/pull/148#issuecomment-861853715>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJDDF3WMTWDAE2AXGZSQ2H3TS7CSPANCNFSM46TE3DJA>
> .
>
